system/service.py

```diff
@@ -520,13 +520,15 @@ def sysv_is_enabled(name):
             return False
 
     def get_systemd_status_dict(self):
-
         # Check status first as show will not fail if service does not exist
         (rc, out, err) = self.execute_command("%s show '%s'" % (self.enable_cmd, self.__systemd_unit,))
-        if rc != 0:
-            self.module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, self.__systemd_unit, err))
-        elif 'LoadState=not-found' in out:
-            self.module.fail_json(msg='systemd could not find the requested service "%r": %s' % (self.__systemd_unit, err))
+        if self.module.params['must_exist']:
+            if rc != 0:
+                self.module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, self.__systemd_unit, err))
+            elif 'LoadState=not-found' in out:
+                self.module.fail_json(msg='systemd could not find the requested service "%r": %s' % (self.__systemd_unit, err))
+        else:
+            self.module.exit_json(changed=False, exists=False)
```

Comment:
I'm assuming from previous code that a not-found case would still exit 0, so I think in both both `must_exist` or not we should catch the `rc != 0` case. Whether or not a not-found scenario leads to a failure should be the only thing inside of the `if self.module.params['must_exist']:` if statement.

As it stands in this change, a run with `must_exist` set to False could simply exit_json even on a non-0 return code.